shell/browser/api/electron_api_session.cc

```diff
@@ -1074,6 +1076,181 @@ std::vector<base::FilePath> Session::GetPreloads() const {
   return prefs->preloads();
 }
 
+/**
+ * Exposes the network service's ClearSharedDictionaryCacheForIsolationKey
+ * method, allowing clearing the Shared Dictionary cache for a given isolation
+ * key. Details about the feature available at
+ * https://developer.chrome.com/blog/shared-dictionary-compression
+ */
+v8::Local<v8::Promise> Session::ClearSharedDictionaryCacheForIsolationKey(
+    const gin_helper::Dictionary& options) {
+  gin_helper::Promise<void> promise(isolate_);
+  auto handle = promise.GetHandle();
+
+  std::string frame_origin_str, top_frame_site_str;
+  if (!options.Get("frameOrigin", &frame_origin_str) ||
+      !options.Get("topFrameSite", &top_frame_site_str)) {
+    promise.RejectWithErrorMessage(
+        "Must provide frameOrigin and topFrameSite strings");
```

Comment:
```suggestion
        "Must provide frameOrigin and topFrameSite strings to `clearSharedDictionaryCacheForIsolationKey`");
```

Big fan of error messages self-declaring where they come from to avoid the classic "what even is an ENOENT" style errors